task/progress.go

```diff
@@ -16,7 +16,9 @@ var progressReportBuckets = []float64{0, 0.25, 0.5, 0.75, 1}
 const minProgressReportInterval = 10 * time.Second
 const progressCheckInterval = 1 * time.Second
 
-func ReportProgress(ctx context.Context, lapi *livepeerAPI.Client, taskID string, size uint64, getCount func() uint64) {
+func ReportProgress(ctx context.Context, lapi *livepeerAPI.Client, taskID string, size uint64, getCount func() uint64, startPercentage int, endPercentage int) {
```

Comment:
small edit: the params can be:

```go
..., startPercentage, endPercentage int)
```
go will assign int type for the `startPercentage`.

Reply:
WDYT of making these floating points instead from 0 to 1, since that's the format of the output of this function? Integers only exist here for the count and total size, but then are immediately converted into floats for the percentage calculation.

It will also make some of the code a little simpler, avoiding a lot of `int()` and `float64()` conversions here.